### DISCUSSION TITLE
uPNP root/home folder

### DISCUSSION THREAD

**Discussion by memascii** [gerbera/gerbera#2837] [Q&A]
Is there a way to configure the root folder which is displayed when connecting with a uPNP client?

What I mean is that if I connect to Gerbera with my Roberts radio, I select "gerbera" but then have to select down the folder structure to get to my music folder, i.e I have to select "PC directory" then "media" then "toshiba" then finally "Music" where my music is stored.  Ideally I would like to just select gerbera and immediately be in my music directory.   Is that possible?  I guess I could maybe do something with a symlink but that doesn't seem right.

TIA and apologies if already covered.
Iain.

**Comment by KarlStraussberger** [comment]
There is no such things as a weird question, but bad docu or even new ideas are more likely. 

Currently, the is no shortcut into the library, but I think we can make up something like "last location" (sounds good to me, if upnp allows).

In the meantime, if you enable virtual layout in config, you get a path like Audio/artist/album.

**Comment by DevFreezing** [comment]
Same issue here, for me makes no sense to show full path when you select a specific folder. I've read the documentation, but there is not a clear way to do this.

**Reply by KarlStraussberger** [reply]
Maybe it can be build on top of the UPnP shortcuts feature (#3261 ) once it's working

**Reply by DevFreezing** [reply]
Thanks for the suggestion, Karl. I tried enabling virtual layout, but it’s not ideal in my case since it relies on file tags, which don’t always align with the directory structure. For many users, especially those organizing files in folders rather than tagging them, having to navigate the full path can be inconvenient. It would be great to have a way to simplify the path display directly in the library without relying on tags.

I'm eagerly awaiting the UPnP shortcuts feature (#3261) once it's implemented—it's something MiniDLNA has had for a while, and it really simplifies the browsing experience!
